### PR TITLE
feat: add config flag to osctl

### LINKED ISSUE #412: Support loading talos config from specified file
It would help (certainly during alpha development) to have a `--talos-config` argument for `osctl` rather than to rely on the predetermined `~/.talos/config`.

### PR DESCRIPTION
**Pull request by andrewrynhard**
Closes #412.

Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

File: internal/app/osctl/cmd/root.go [hunk 1]
```diff
@@ -34,6 +37,11 @@ var rootCmd = &cobra.Command{
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	u, err := user.Current()
+	if err != nil {
+		return
+	}
+	rootCmd.PersistentFlags().StringVar(&config, "talosconfig", path.Join(u.HomeDir, ".talos", "config"), "The path to the Talos configuration file")
```
Comment (andrewrynhard) [inline internal/app/osctl/cmd/root.go]: `osctl --talosconfig`

**Comment by bradbeam** [comment]
`talos` config is the same as the userdata correct?
in #368 i'm referring to this as `-u // --userdata`, we probably should make this consistent from a UX perspective. I'm not too particular on the name of it.

**Comment by andrewrynhard** [comment]
@bradbeam it is not. This is the local config for use with `osctl`. `--userdata` works for the node config. It aligns with what is in the code.

**Comment by andrewrynhard** [comment]
```
[I] $ ./build/osctl-linux-amd64 --help
A CLI for out-of-band management of Kubernetes nodes created by Talos
...
Flags:
  -h, --help                 help for osctl
      --talosconfig string   The path to the Talos configuration file (default "/root/.talos/config")
```

**Pull request merged**